File: cpp_tests/test_statistik.cpp
#include "tests/test_statistik.hpp"

void TestStatistik() {
  ReadData inventur{"./data/supermarkt_1/produkte.txt"};
  std::string data = inventur.ReadTxt();

  ReadData kunden{"./data/supermarkt_0/kunden.txt"};
  std::string kundenData = kunden.ReadTxt();

  ReadData verkaeufer{"./data/supermarkt_0/haendler.txt"};
  std::string verkaeuferData = verkaeufer.ReadTxt();

  ReadData warenkoerbe{"./data/supermarkt_1/warenkoerbe.txt"};
  std::string warenkoerbeData = warenkoerbe.ReadTxt();

  ReadData settings{"./.vscode/settings.json"};
  std::string settingsData = settings.ReadJSON();

  ReadData numbers{"./data/numbers.csv"};
  std::string numbersData = numbers.ReadCSV();

  ReadData pdf{"README.pdf"};
  std::string pdfData = pdf.ReadPDF();

  Statistik statistik;
  statistik.LadeDaten(warenkoerbe);
  assert(statistik.ZaehleEintraege() == 1);
  assert(statistik.ZaehleVariable("DATABASE") == 1);
  statistik.EntferneDaten(warenkoerbeData);
  assert(statistik.ZaehleEintraege() == 0);
  statistik.EntferneDaten(warenkoerbeData);
  assert(statistik.ZaehleEintraege() == 0);
  statistik.LadeDaten(inventur);
  assert(statistik.ZaehleEintraege() == 1);
  assert(statistik.ZaehleLinien() == 11);
  assert(statistik.ZaehleWorte() == 14);
  statistik.LadeDaten(numbers);
  assert(statistik.ZaehleEintraege() == 2);
  assert(statistik.ZaehleLinien() == 54);
  assert(statistik.ZaehleWorte() == 14);
  assert(statistik.ZaehleZeichen() == 2084);
  assert(std::fabs(statistik.Mittelwert() - 197.14f) < 0.01f);
  assert(std::fabs(statistik.Median() - 1042.00f) < 0.01f);
  assert(std::fabs(statistik.Erwartungswert() - 197.14f) < 0.01f);
  assert(std::fabs(statistik.Standardabweichung() - 117.26f) < 0.01f);

  std::cout << "Alle Statistik-Tests waren erfolgreich!" << std::endl;
}

File: python_tests/test_statistik.py
"""Python-file for testing class Statistik"""

from py_bindings import ReadData, Statistik

def test_statistik():
    """Funktion zum Testen von Statistik"""
    text = ReadData("./data/supermarkt_0/produkte.txt")
    kunden_data = ReadData("./data/supermarkt_0/kunden.txt")

    statistik = Statistik()
    statistik.lade_daten(text)
    statistik.entferne_daten(text.read_txt())
    statistik.lade_daten(text)
    assert statistik.zaehle_linen() == 11
    assert statistik.wortanzahl() in (14, 17)
    assert statistik.zaehle_variable("Produkt") == 1
    assert statistik.zaehle_eintraege() == 1
    assert statistik.zeichenanzahl() in (596, 598)
    statistik.lade_daten(kunden_data)
    assert statistik.zaehle_linen() == 22
    assert statistik.wortanzahl() in (33, 38, 41, 40)
    assert statistik.zaehle_variable("Kunde") == 1
    assert statistik.zaehle_eintraege() == 2
    assert 7.508571147 - statistik.mittelwert() < 0.0001
    assert statistik.zeichenanzahl() in (1206, 1207, 1208, 1209)
    assert 601.00001321 - statistik.median() < 0.0001
    assert 7.5085711479187012 - statistik.erwartungswert() < 0.0001
    assert 12.2170724868774412 - statistik.standardabweichung() < 0.0001

test_statistik()
